src/mc_cell_group.hpp

```diff
@@ -0,0 +1,272 @@
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <iterator>
+#include <vector>
+
+#include <algorithms.hpp>
+#include <cell_group.hpp> // CHANGE
```

Comment:
Good spot. The `CHANGE` comments were placeholders that I was using during development, which I forgot to remove. I will remove them.